pkg/auth/oauth/external/github/github_test.go

```diff
@@ -7,5 +7,5 @@ import (
 )
 
 func TestGitHub(t *testing.T) {
-	_ = external.Provider(NewProvider("github", "clientid", "clientsecret"))
+	_ = external.Provider(NewProvider("github", "clientid", "clientsecret", nil))
```

Comment:
This looks like a  very useful test.  I'd rather have the type assertion in the code.